plugins/destination/snowflake/client/read.go

```diff
@@ -24,67 +24,112 @@ func (c *Client) reverseTransform(f arrow.Field, bldr array.Builder, val any) er
 	}
 	switch b := bldr.(type) {
 	case *array.BooleanBuilder:
-		b.Append(val.(bool))
+		if boolVal, ok := val.(bool); ok {
+			b.Append(boolVal)
+		} else {
+			boolVal, err := strconv.ParseBool(val.(string))
+			if err != nil {
+				return fmt.Errorf("failed to parse bool: %w", err)
+			}
+			b.Append(boolVal)
+		}
 	case *array.Int8Builder:
 		u, err := strconv.ParseInt(val.(string), 10, 8)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to parse int8: %w", err)
 		}
 		b.Append(int8(u))
 	case *array.Int16Builder:
 		u, err := strconv.ParseInt(val.(string), 10, 16)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to parse int16: %w", err)
 		}
 		b.Append(int16(u))
 	case *array.Int32Builder:
 		u, err := strconv.ParseInt(val.(string), 10, 32)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to parse int32: %w", err)
 		}
 		b.Append(int32(u))
 	case *array.Int64Builder:
 		u, err := strconv.ParseInt(val.(string), 10, 64)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to parse int64: %w", err)
 		}
 		b.Append(u)
 	case *array.Uint8Builder:
 		u, err := strconv.ParseUint(val.(string), 10, 8)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to parse uint8: %w", err)
 		}
 		b.Append(uint8(u))
 	case *array.Uint16Builder:
 		u, err := strconv.ParseUint(val.(string), 10, 16)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to parse uint16: %w", err)
 		}
 		b.Append(uint16(u))
 	case *array.Uint32Builder:
 		u, err := strconv.ParseUint(val.(string), 10, 32)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to parse uint32: %w", err)
 		}
 		b.Append(uint32(u))
 	case *array.Uint64Builder:
 		u, err := strconv.ParseUint(val.(string), 10, 64)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to parse uint64: %w", err)
 		}
 		b.Append(u)
 	case *array.Float32Builder:
-		b.Append(val.(float32))
+		if floatVal, ok := val.(float64); ok {
+			b.Append(float32(floatVal))
+		} else {
+			floatVal, err := strconv.ParseFloat(val.(string), 32)
+			if err != nil {
+				return fmt.Errorf("failed to parse float32: %w", err)
+			}
+			b.Append(float32(floatVal))
+		}
 	case *array.Float64Builder:
-		b.Append(val.(float64))
+		if floatVal, ok := val.(float64); ok {
+			b.Append(floatVal)
+		} else {
+			floatVal, err := strconv.ParseFloat(val.(string), 64)
+			if err != nil {
+				return fmt.Errorf("failed to parse float64: %w", err)
+			}
+			b.Append(floatVal)
+		}
 	case *array.StringBuilder:
 		b.Append(val.(string))
 	case *array.LargeStringBuilder:
 		b.Append(val.(string))
 	case *array.BinaryBuilder:
 		b.Append(val.([]uint8))
 	case *array.TimestampBuilder:
-		b.Append(arrow.Timestamp(val.(time.Time).UnixMicro()))
+		var timeVal time.Time
```

Comment:
https://pkg.go.dev/github.com/apache/arrow/go/v13@v13.0.0-20230516163220-505a2e4519ba/arrow#TimestampFromString
+ `b.Type().(*arrow.TimestampType).Unit`

Reply:
not sure I understand this comment.

Reply:
Instead of parsing manually you could use `arrow.TimestampFromString(val.(string), b.Type().(*arrow.TimestampType).Unit`